src/panels/lovelace/cards/hui-alarm-panel-card.ts

```diff
@@ -0,0 +1,313 @@
+interface Config extends LovelaceCardConfig {
+  entity: string;
+  name?: string;
+  states?: string[];
+}
+
+class HuiAlarmPanelCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+  private _code?: string;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+      _code: {},
+    };
+  }
+
+  public getCardSize(): number {
+    return 4;
+  }
+
+  public setConfig(config: Config): void {
+    if (
+      !config ||
+      !config.entity ||
+      config.entity.split(".")[0] !== "alarm_control_panel"
+    ) {
+      throw new Error("Invalid card configuration");
+    }
+
+    const defaults = {
+      states: ["arm_away", "arm_home"],
+    };
+
+    this._config = { ...defaults, ...config };
+    this._code = "";
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    if (changedProps.has("_config")) {
+      return true;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+    if (oldHass) {
+      return (
+        oldHass.states[this._config!.entity] !==
+        this.hass!.states[this._config!.entity]
+      );
+    }
+    return true;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+    const stateObj = this.hass.states[this._config.entity];
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card
+        .header="${this._config.name || this._label(stateObj.state)}"
+        class="${classMap({ "not-found": !stateObj })}"
+      >
+        ${
+          !stateObj
+            ? html`
+                <div class="not-found">
+                  Entity not available: ${this._config.entity}
+                </div>
+              `
+            : html`
+                <ha-label-badge
+                  class="${classMap({ [stateObj.state]: true })}"
+                  .icon="${ICONS[stateObj.state] || "hass:shield-outline"}"
+                  .label="${this._stateIconLabel(stateObj.state)}"
+                ></ha-label-badge>
+                ${
+                  stateObj.state === "disarmed" && this._config!.states
+                    ? html`
+                        <div id="armActions" class="actions">
+                          ${
+                            this._config.states!.map((state) => {
+                              return html`
+                                <paper-button
+                                  noink
+                                  raised
+                                  .action="${state}"
+                                  @click="${this._handleActionClick}"
+                                  >${this._label(state)}</paper-button
+                                >
+                              `;
+                            })
+                          }
+                        </div>
+                      `
+                    : html`
```

Comment:
Since this is so similar to the other rendering of the buttons, why don't we just do this:

```ts
(this._config.states || ["disarm"]).map((state) => {
```

That should give the same results?

Reply:
States will still be available if you only want only disarm to show